lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -803,6 +807,41 @@ def accessible_objects(user_or_attribute, operation, options = {})
       end
     end
 
+    # given a user / user attribute and a list object / object attribute ids
+    # return a map of object attribute ids to all operation names between each of them and the given user / user attribute
+    # all operations names means for both positive operations and negative operations (prohibitions)
+    def all_operations_for_user_or_attr_and_objs_or_attrs(user_or_attribute, object_attribute_ids, options)
+      return {} if object_attribute_ids.empty?
+
+      opset_ids_and_obj_attr_ids = associations_for_user_or_attribute(user_or_attribute, options).where(
+        object_attribute_id: object_attribute_ids
+      ).pluck(:operation_set_id, :object_attribute_id)
+
+      # generate a hash of object attribute ids to lists of operation sets from their
+      # associations with the given user / user attribute
+      obj_attr_ids_to_opset_ids = opset_ids_and_obj_attr_ids.each_with_object(
+        Hash.new { |h, k| h[k] = [] }
+      ) do |(opset_id, objattr_id), acc|
+        acc[objattr_id] << opset_id
+      end
```

Comment:
```suggestion
      obj_attr_ids_to_opset_ids = Hash[*opset_ids_and_obj_attr_ids.flatten]
```
I think this does the same thing.

Reply:
or just `Array#to_h`

Reply:
I missed that the value is wrapped in an array

```Ruby
obj_attr_ids_to_opset_ids = opset_ids_and_obj_attr_ids..to_h { |k, v| [k, [v]] }
```

Reply:
the prob here that we can  have multiple opset ids to the same obj attr id.. so  like:

```ruby
[['opset1_id',  'obj_attr1_id'], ['opset2_id', 'obj_attr1_id']]
```

we want:

```ruby
{ 'obj_attr1_id' =>  ['opset1_id', 'opset2_id'] }
```

i think ur snippet will give:

```ruby
{
  'opset1_id' =>  ['obj_attr1_id'],
  'opset2_id' => ['obj_attr1_id'],
}
```

Reply:
ah ok, I got twisted reading the variable names.